invenio_banners/services/service.py

```diff
@@ -40,16 +50,49 @@ def read_active_banners(self, identity, url_path):
             links_item_tpl=self.links_item_tpl,
         )
 
-    def search(self, identity, limit=100, params=None):
+    def search(self, identity, params):
         """Search for banners matching the querystring."""
         self.require_permission(identity, "search")
 
-        banners = self.record_cls.search(limit, params)
+        search_params = get_search_params(self.config, params)
+
+        query_param = search_params["q"]
+        filters = []
+
+        if query_param:
```

Comment:
minor: I'm a bit sceptical about implementing our own custom query-parsing logic here, though I understand the need since we can't just pass a user-provided SQL query or similar...

Since the most familiar query syntax we're using across Invenio is actually the ElasticSearch/Lucene query string syntax, it could make that sense that we make a light "translator" to the equivalent SQLAlchemy query? We could use something like [`luqum`](https://luqum.readthedocs.io/en/latest/quick_start.html) (already used for rewriting ES queries in other modules).

This might stretch the scope of the task by a lot though, so not sure where to draw the line. @kpsherva @zzacharo @ntarocco WDYT?

Reply:
We have discussed this and we agree with you. The main point here was that it will take time and effort to create that and hopefully we don't implement many searches in the DB, only special cases.
I think it is for another task, and for now simply emulate what was done with OAI-PMH, WDYT?

For example, I don't see the point of creating queries for banner with date ranges `from TO to` or more advanced cases...